raiden/raiden_event_handler.py

```diff
@@ -603,104 +612,217 @@ def handle_contract_send_channelunlock(
-                # In that case it's safe to just use the latest view of the chain to
-                # query the on-chain participant/channel details
-                triggered_by_block_hash = token_network_proxy.client.blockhash_from_blocknumber(
-                    "latest"
-                )
-            participants_details = token_network_proxy.detail_participants(
+        """ Handles settlement for normal node. """
+
+        settlement_parameters = RaidenEventHandler.process_data_and_get_settlement_parameters(
+            raiden,
+            channel_settle_event.token_network_identifier,
+            channel_settle_event.channel_identifier,
+            channel_settle_event.triggered_by_block_hash
+        )
+
+        canonical_identifier = CanonicalIdentifier(
+            chain_identifier=raiden.chain.network_id,
+            token_network_address=channel_settle_event.token_network_identifier,
+            channel_identifier=channel_settle_event.channel_identifier,
+        )
+
+        payment_channel: PaymentChannel = raiden.chain.payment_channel(
+            canonical_identifier=canonical_identifier
+        )
+
+        payment_channel.settle(
+            transferred_amount=settlement_parameters.transferred_amount,
+            locked_amount=settlement_parameters.locked_amount,
+            locksroot=settlement_parameters.locksroot,
+            partner_transferred_amount=settlement_parameters.partner_transferred_amount,
+            partner_locked_amount=settlement_parameters.partner_locked_amount,
+            partner_locksroot=settlement_parameters.partner_locksroot,
+            block_identifier=settlement_parameters.block_identifier,
+        )
+
+    @staticmethod
+    def handle_contract_send_channel_settle_light(raiden: "RaidenService",
+                                                  channel_settle_light_event: ContractSendChannelSettleLight):
+        """ Store a message for the LC with SettlementRequired type to handle settlement for LC on hub mode. """
+
+        log.debug("Handling channel settle light")
+
+        settlement_parameters = RaidenEventHandler.process_data_and_get_settlement_parameters(
+            raiden,
+            channel_settle_light_event.token_network_identifier,
+            channel_settle_light_event.channel_identifier,
+            channel_settle_light_event.triggered_by_block_hash)
+
+        canonical_identifier = CanonicalIdentifier(
+            chain_identifier=raiden.chain.network_id,
+            token_network_address=channel_settle_light_event.token_network_identifier,
+            channel_identifier=channel_settle_light_event.channel_identifier,
+        )
+        payment_channel: PaymentChannel = raiden.chain.payment_channel(
+            canonical_identifier=canonical_identifier
+        )
+
+        pseudo_random_generator = random.Random()
+
+        message_identifier = message_identifier_from_prng(pseudo_random_generator)
+
+        # and now find out
```

Comment:
incomplete comment?

Reply:
Exctracted code comment, doesn't apply anymore, but i will check just in case

Reply:
Done